Assets/Implementation/Pathfinding/Scripts/NPCController.cs

```diff
@@ -0,0 +1,213 @@
+        NativeHashMap<Vector3Int, Node> _openList =
+            new NativeHashMap<Vector3Int, Node>(arraySize / 2, Allocator.TempJob);
+        NativeHashMap<Vector3Int, Node> _closedList =
+            new NativeHashMap<Vector3Int, Node>(arraySize / 2, Allocator.TempJob);
+        NativeArray<Vector3Int> _neighbourOffsets = new NativeArray<Vector3Int>(8, Allocator.TempJob);
+        NativeArray<Node> _backtrackedPath = new NativeArray<Node>(arraySize, Allocator.TempJob);
+        NativeArray<int> _backtrackedPathLength = new NativeArray<int>(1, Allocator.TempJob);
+
+        // Add all nodes in the generated grid to the NativeHashMap (Replace this with cody's grid system)
+        foreach (var item in _nodeGrid)
+        {
+            _gridNodes.Add(item.position, item);
+        }
+
+        // Create a new instance of the AStar job and assign its variables
+        AStar aStar = new AStar
+        {
+            gridNodes = _gridNodes,
+            openList = _openList,
+            closedList = _closedList,
+            neighbourOffsets = _neighbourOffsets,
+            startNode = _nodeGrid[(int)transform.position.x, (int)transform.position.y, (int)transform.position.z],
+            endNode = _nodeGrid[destination.x, destination.y, destination.z],
+            backtrackedPath = _backtrackedPath,
+            backTrackedPathLength = _backtrackedPathLength
+        };
+
+        // Schedule the job and complete it
+        JobHandle handle = aStar.Schedule();
+        handle.Complete();
+
+        // retrieve the backtracked path from the job
+        for (int i = 0; i < _backtrackedPathLength[0]; i++)
+        {
+            this._backtrackedPath.Add(_backtrackedPath[i]);
+        }
+
+        open = _openList.GetValueArray(Allocator.Temp).ToArray();
+        closed = _closedList.GetValueArray(Allocator.Temp).ToArray();
+
+        // Dispose all the NativeContainers to avoid memory leaks
+        _gridNodes.Dispose();
+        _openList.Dispose();
+        _closedList.Dispose();
+        _neighbourOffsets.Dispose();
+        _backtrackedPath.Dispose();
+        _backtrackedPathLength.Dispose();
+    }
+
+    /// <summary>
+    /// Create a grid of nodes
+    /// </summary>
+    void CreateGrid()
+    {
+        _nodeGrid = new Node[_gridWidth, _gridHeight, 1];
+
+        var untraversable = _grid.UnTraversable();
+        for (int x = 0; x < _gridWidth; x++)
+        {
+            for (int y = 0; y < _gridHeight; y++)
+            {
+                Node node = new Node()
+                {
+                    position = new Vector3Int(x, y, 0),
+                    traversable = !untraversable[x, y]
+                };
+                _nodeGrid[x, y, 0] = node;
+            }
+        }
+    }
+
+    public float RoundToMultiple(float value, float roundTo)
+    {
+        return Mathf.RoundToInt(value / roundTo) * roundTo;
+    }
+
+    private void OnDrawGizmosSelected()
+    {
+        if (!_drawGizmos) return;
```

Comment:
I don't see the point of having a bool that determines whether or not gizmos are drawn when the only moment they're drawn is when the object is selected...